src/master-playlist-controller.js

```diff
@@ -361,6 +370,32 @@ export class MasterPlaylistController extends videojs.EventTarget {
         } else {
           addSeekableRange();
         }
+
+        buffered = this.tech_.buffered();
+        currentTime = this.tech_.currentTime();
+
+        if (buffered.length) {
+          lastBufferedEnd = buffered.end(buffered.length - 1);
+        }
+        bufferedTime = lastBufferedEnd - currentTime;
+
+        // the end of the current playlist
+        playEnd = this.playlistEnd(updatedPlaylist);
+
+        // the time between end of the playlist and the end of the buffered
+        endTime = playEnd - lastBufferedEnd;
+
+        if (this.seekable().length) {
+          seekableEnd = this.seekable().end(0);
+        }
+
+        // blacklist this playlist if the playhead is at the end of the buffer
+        // and the buffered data ends at the end of the last segment in the playlist
+        if (bufferedTime <= Ranges.TIME_FUDGE_FACTOR &&
+            endTime <= Ranges.TIME_FUDGE_FACTOR &&
+            currentTime > seekableEnd) {
```

Comment:
So I was giving this more thought and I think we can avoid worrying about it all together by calling this method and blacklisting the playlist outside of the `loadedplaylist` listener. Playlist Loader already knows when a playlist it has loaded [didn't actually update the live window](https://github.com/videojs/videojs-contrib-hls/blob/master/src/playlist-loader.js#L188). The `isPlaylistOutdated_` method more so checks if the player is stuck at the end despite loading "fresh" playlists. If instead playlist loader triggered an event at that moment it knows the playlist isnt a fresh update, then `MasterPlaylistController` can listen to that and run the checks for playlistOutdated and if those checks pass, then we blacklist. This way we wont run these checks every time the playlist is refreshing correctly and run the risk of accidentally blacklisting a playlist we don't want to